### ISSUE TITLE
Filtering Ingresses by class

### ISSUE THREAD

**Issue by cubic3d** [ori-edge/k8s_gateway#36]
Currently it's impossible to filter which Ingress class should be taken into account for polling host names and Loadbalancer IPs.
Example: Two ingress-controllers `nginx-int` and `nginx-ext` each having an Ingress resource defining `host.domain.tld` - the resulting response to the query looks like that:
```
;; ANSWER SECTION:
host.domain.tld.	60	IN	A	192.168.99.2
host.domain.tld.	60	IN	A	192.168.99.1
```
But from the internal network (responsibility of the `nginx-int` controller) only one of those Loadbalancer IPs is accessible on the default port without IP restrictions which leads to timeouts and inability to access the service. Filtering based on `ingressClassName` and `kubernetes.io/ingress.class` or some sort of `coredns.io/ignore` annotation is required to solve that.

**Comment by networkop** [comment]
yeah, it makes sense. The original CoreDNS issue had a [suggestion](https://github.com/coredns/coredns/issues/2465#issuecomment-593910983) for how this could be done. PRs are welcome

**Comment by cubic3d** [comment]
I still have an eye on this, just had no time yet ;)
